sample.shell.shell_module.minimal_rtt fails to build on mimxrt1170_evk_cm4/mimxrt1170_evk_cm7

`./scripts/twister -p mimxrt1170_evk_cm4 -p mimxrt1170_evk_cm7 -s samples/subsys/shell/shell_module/sample.shell.shell_module.minimal_rtt`

Fails as follows:

```
[ 91%] Linking C static library lib..__modules__hal__nxp.a
[ 91%] Built target ..__modules__hal__nxp
Scanning dependencies of target modules__segger
[ 92%] Building C object modules/segger/CMakeFiles/modules__segger.dir/home/galak/git/modules/debug/segger/SEGGER/SEGGER_RTT.c.obj
[ 92%] Building C object modules/segger/CMakeFiles/modules__segger.dir/SEGGER_RTT_zephyr.c.obj
[ 93%] Linking C static library libmodules__segger.a
[ 93%] Built target modules__segger
Scanning dependencies of target zephyr_prebuilt
[ 94%] Building C object zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj
[ 95%] Linking C executable zephyr_prebuilt.elf
/opt/zephyr-sdk-0.13.0-alpha-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `.dtcm_data' from `../modules/segger/libmodules__segger.a(SEGGER_RTT.c.obj)' being placed in section `.dtcm_data'
collect2: error: ld returned 1 exit status
gmake[2]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:122: zephyr/zephyr_prebuilt.elf] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3062: zephyr/CMakeFiles/zephyr_prebuilt.dir/all] Error 2
gmake: *** [Makefile:103: all] Error 2
```

--- a/soc/arm/nxp_imx/rt/Kconfig.defconfig.series
+++ b/soc/arm/nxp_imx/rt/Kconfig.defconfig.series
@@ -136,13 +136,15 @@ config DMA_MCUX_EDMA
 	default y if HAS_MCUX_EDMA
 	depends on DMA
 
+DT_CHOSEN_Z_DTCM := zephyr,dtcm
+
 choice SEGGER_RTT_SECTION
-	default SEGGER_RTT_SECTION_DTCM if HAS_MCUX_SEMC
+	default SEGGER_RTT_SECTION_DTCM if $(dt_chosen_enabled,$(DT_CHOSEN_Z_DTCM))
 	depends on USE_SEGGER_RTT
 endchoice
 
 choice SEGGER_SYSVIEW_SECTION
-	default SEGGER_SYSVIEW_SECTION_DTCM if HAS_MCUX_SEMC
+	default SEGGER_SYSVIEW_SECTION_DTCM if $(dt_chosen_enabled,$(DT_CHOSEN_Z_DTCM))
 	depends on SEGGER_SYSTEMVIEW
 endchoice
 

